maven-core/src/main/java/org/apache/maven/DefaultMaven.java

```diff
@@ -382,34 +397,50 @@ private void validateLocalRepository( MavenExecutionRequest request )
                 logger.warn( "Failed to lookup lifecycle participants: " + e.getMessage() );
             }
 
-            Collection<ClassLoader> scannedRealms = new HashSet<>();
+            lifecycleListeners.addAll( getProjectScopedExtensions( projects,
+                                                                   AbstractMavenLifecycleParticipant.class ) );
+        }
+        finally
+        {
+            Thread.currentThread().setContextClassLoader( originalClassLoader );
+        }
```

Comment:
The classloader is never overridden here, leftover?

Reply:
Done